### DISCUSSION TITLE
Can't create variables in GraphQl

### DISCUSSION THREAD

**Discussion by parich15** [directus/directus#4015] [Archive]
Hello and thanks for your time in advance. 
I'm a newbie in Directus, but when it comes to Headless CMS,  I've worked with strapi and gridsome so it's not my first time developing this kind of apps. 

The thing is, i'm creating a Nuxt website and I'm using Directus 9.0.0-rc.24 for the backend. (Running on a docker container with PostgreSQL via docker-compose from the Directus docs.) I've already created my data collections and connected the endpoints to my site using Apollo because I'm using GraphQl to make the queries. 

I need to pass a variable to get the id of the post to generate a dynamic route based on this id. For that, i was using GraphQl variables like always but this time it seems impossible. I've readed the docs, the discussions from here in github and stack overflow but i can't find anything that helps me. 

I will attach you the pics of my queries and what I'm trying to achieve, if anybody knows how can I get the id i will be very grateful. 

1// The first is what I'm trying to do. 
2// Is what im'used to do when it comes to variables in graphql
3// If I remove the Variable code it works.
![Captura](https://user-images.githubusercontent.com/60244188/107659729-bd76db00-6c87-11eb-8f80-cedc5604605c.PNG)
![Captura2](https://user-images.githubusercontent.com/60244188/107659730-bd76db00-6c87-11eb-8cdb-53f2b3988e42.PNG)
![Captura3](https://user-images.githubusercontent.com/60244188/107659727-bcde4480-6c87-11eb-86a1-1f15f6eae882.PNG)

**Comment by rijkvanzanten** [comment]
Hi @parich15!

There is no `id` query attribute available in GraphQL. See https://docs.directus.io/reference/api/query/ for the list of available attributes.

**Reply by parich15** [reply]
Wow that was fast, thanks for the answer!😄 So that means there is no way to get the Id or any other Variable for generating the route? 
Ps: keep up the good work, developers like you are meant to be a reference 🔥😎

**Reply by rijkvanzanten** [reply]
Variables work just as expected 🙂 Just saying that you're trying to use an attribute `id` that doesn't exist:

```graphql
query {
  items {
    pages(id: 15) {
          ^^ id doesn't exist here
      id
    }
  }
}
```

Any of the existing attributes work with variables as well, for example:

```graphql
query getPage($limit: Int) {
  items {
    pages(limit: $limit) {
      id
    }
  }
}
```

If you wanna filter by ID, use [`filter`](https://docs.directus.io/reference/api/query/#filter) instead:

```graphql
query getPage($id: ID) {
  items {
    pages(filter: { id: { _eq: $id }}) {
      id
    }
  }
}
```

**Reply by parich15** [reply]
Thanks a lot again for your answer, I will try to figure out how to do it because I'm sure is possible, and in the process learn more and more. If I achieve something I will post it later! Have a nice day!
